pam/pam-module-transaction-dummy.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+)
+
+// PamModuleTransactionDummy is an implementation of PamModuleTransaction for
+// testing purposes.
+type PamModuleTransactionDummy struct {
```

Comment:
As this type has some implementations (i.e. methods), I would name it PamModuleTransactionMock instead.
Or, even better, after switching the package to `pam` rather than `main` (see my comment on it), it's better to also remove the `Pam` prefix, as usage would be `pam.ModuleTransactionMock`.

Reply:
Fair enough.

As per the `Mock` suffix, it was my initial choice, then I thought that we could have used `gomock` for some other testing, and that implementation would have been a real mock, while this one is more a basic implementation to be used for testing some basic behaviors.